src/main/webapp/WEB-INF/jsp/upliftRegister.jsp

```diff
@@ -51,51 +75,81 @@
 
                     <h2 class="heading-medium"><spring:message code="public.uplift.user.body" /></h2>
 
-                    <div class="form-group ${isFirstNameEmpty? 'form-group-error' : ''}">
-                        <label for="firstName">
-                            <span class="form-label-bold"><spring:message code="public.uplift.user.first.name.label" /></span>
-                            <c:if test="${isFirstNameEmpty}">
-                                <script>
-                                    sendEvent('Uplift Registration', 'Error', 'First name is empty');
-                                </script>
-                                <span class="error-message"><spring:message code="public.common.error.empty.first.name" /></span>
-                            </c:if>
-                        </label>
-                        <input class="form-control form-control-3-4 ${isFirstNameEmpty? 'form-control-error' : ''}" type="text" id="firstName" name="firstName" value="${fn:escapeXml(param['firstName'])}" autocomplete="off">
-                    </div>
+                    <spring:bind path="firstName">
+                        <div class="form-group ${status.error ? 'form-group-error' : ''}">
+                            <label for="firstName">
+                                <span class="form-label-bold"><spring:message code="public.self.register.first.name.label"/></span>
+                                <c:if test="${status.error}">
+                                        <span class="error-message">
+                                            <ul>
+                                                <c:forEach var="error" items="${status.errorCodes}">
+                                                    <li><spring:message code="${error}${'.selfRegisterCommand.firstName'}"></spring:message></li>
+                                                    <script>
+                                                        sendEvent('Uplift Register', 'Error', 'First name error code: ${status.errorCode}');
```

Comment:
not sure we should change this as it won't match previous analytics data and might affect some dashboards and such